src/firmware/linux/host/types.rs

```diff
@@ -356,54 +377,59 @@ impl CertTable {
     /// };
     /// ```
     ///
-    unsafe fn parse_table(&self) -> Vec<UapiCertTableEntry> {
-        const OFFSET_START: isize = (std::mem::size_of::<u8>() * 16) as isize;
-        const LENGTH_START: isize = (std::mem::size_of::<u8>() * 20) as isize;
-        const ENTRY_SIZE: isize = (std::mem::size_of::<u8>() * 24) as isize;
-        const GUID_SIZE: usize = 16_usize;
+    pub unsafe fn parse_table(mut data: *mut CertTableEntry) -> Vec<UapiCertTableEntry> {
+        // Helpful Constance for parsing the data
+        const ZERO_GUID: Uuid = Uuid::from_bytes([0x0; 16]);
+
+        // Pre-defined re-usable variables.
+        let mut guid: Uuid;
+        let table_ptr: *mut u8 = data as *mut u8;
+        let mut cert_addr: *mut u8;
+        let mut cert_end: *mut u8;
+        let mut cert_bytes: Vec<u8> = vec![];
 
+        // Create a location to store the final data.
         let mut retval: Vec<UapiCertTableEntry> = vec![];
-        let mut guid_ptr: *const CertTableEntry = self.entries;
-        let mut guid: String;
-        let mut cert_address: *mut u8;
-        let mut cert_len: usize;
-        let mut cert_end: *mut u8;
+
+        // Start parsing the PSP data from the pointers.
+        let mut entry: CertTableEntry;
 
         loop {
-            let mut cert_bytes: Vec<u8> = vec![];
+            // Dereference the pointer to parse the table data.
+            entry = *data;
+            guid = Uuid::from_slice(entry.guid.as_slice()).unwrap();
```

Comment:
Consider `?` instead of `.unwrap()` -- not sure that a parsing problem should panic the entire program. This change will require returning a `Result<...>`.

Reply:
TY. Covered in next commit.